### PR TITLE
Correctly dereference stored images when leaving channels

### LINKED ISSUE #1404: Stored images are not dereferenced when leaving a channel
... or closing network, or closing the demo page.

### PR DESCRIPTION
**Pull request by xPaw**
Fixes #1404. cc @MaxLeiter

### CODE REVIEW

File: src/plugins/storage.js [hunk 1]
```diff
@@ -17,12 +17,6 @@ class Storage {
 	}
 
 	dereference(url) {
-		// If maxHistory is 0, image would be dereferenced before client had a chance to retrieve it,
-		// so for now, just don't implement dereferencing for this edge case.
-		if (helper.maxHistory === 0) {
```
Comment (xPaw) [inline src/plugins/storage.js]: There was a bug here too, should have been `helper.config.maxHistory`.

Comment (astorije) [inline src/plugins/storage.js]: Just curious, why did you move it out of the function to somewhere else?

Comment (xPaw) [inline src/plugins/storage.js]: I didn't feel like adding a `force` option to avoid this, so I just moved it instead.

Comment (astorije) [inline src/plugins/storage.js]: Sorry, what I meant was: why is this needed in one case and not every time you call `dereference`?

Comment (xPaw) [inline src/plugins/storage.js]: Other cases are leaving channel/network/closing public page, which all destroy the channel and messages completely.
